pkg/logs/logs_test.go

```diff
@@ -33,10 +35,25 @@ import (
 func TestLogs(t *testing.T) {
 	if flags, found := os.LookupEnv("GO_CHILD_FLAG"); found {
 		if _, found := os.LookupEnv("GO_CHILD_SKIP_INITIALIZE"); !found {
-			fs := pflag.NewFlagSet("test-logs", pflag.ExitOnError)
+			fs := pflag.NewFlagSet("test-logs", pflag.ContinueOnError)
+			fs.SetOutput(io.Discard)
 			logs.AddFlags(fs)
-			fs.Parse(strings.Split(flags, " "))
-			logs.Initialize()
+			if err := fs.Parse(strings.Split(flags, " ")); err != nil {
+				exitCode := 0
+				if errors.Is(err, pflag.ErrHelp) {
+					fmt.Fprint(os.Stdout, fs.FlagUsages())
+					os.Exit(exitCode)
+				} else {
+					exitCode := 1
+					klog.ErrorS(err, "Exiting due to error", "exit-code", exitCode)
+					klog.FlushAndExit(time.Second, exitCode)
+				}
+			}
+			if err := logs.Initialize(); err != nil {
+				exitCode := 1
+				klog.ErrorS(err, "Exiting due to error", "exit-code", exitCode)
+				klog.FlushAndExit(time.Second, exitCode)
+			}
 		}
```

Comment:
These changes are intended to make the logs test fixture parse the flags and behave the same way as Cobra.
* --help / -h results in stdout and exit code 0
* Unrecognized flags result in an error which is logged to stderr with exit code 1

Reply:
Thanks for the extra context. I think it would be worth adding a comment above to keep that in mind in the future. I'm already imagining myself in a year: why do we do this weird dance with the exit codes in the test code?

Reply:
I'll merge this now and add the comment in one of the followup PRS.